config.go

```diff
@@ -159,57 +173,90 @@ func (ca *Cagent) userAgent() string {
 
 func (ca *Cagent) DumpConfigToml() string {
 	buff := &bytes.Buffer{}
-	enc := toml.NewEncoder(buff)
-	err := enc.Encode(ca)
 
+	err := toml.NewEncoder(buff).Encode(ca)
 	if err != nil {
 		log.Errorf("DumpConfigToml error: %s", err.Error())
+		return ""
 	}
 
 	return buff.String()
 }
 
-func (ca *Cagent) ReadConfigFromFile(configFilePath string, createIfNotExists bool) error {
-	dir := filepath.Dir(configFilePath)
-	err := os.MkdirAll(dir, 0755)
+func (ca *Cagent) ReadConfigFromFile(configFilePath string) error {
+	_, err := os.Stat(configFilePath)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to create the config dir: '%s'", dir)
+		return err
 	}
 
-	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
-		if !createIfNotExists {
-			return fmt.Errorf("Config file not exists: %s", configFilePath)
-		}
-		f, err := os.OpenFile(configFilePath, os.O_WRONLY|os.O_CREATE, 0644)
+	_, err = toml.DecodeFile(configFilePath, &ca)
 
-		if err != nil {
-			return fmt.Errorf("Failed to create the default config file: '%s'", configFilePath)
-		}
-		defer f.Close()
-		enc := toml.NewEncoder(f)
-		enc.Encode(ca)
-	} else {
-		_, err = os.Stat(configFilePath)
-		if err != nil {
-			return err
+	ca.lookupEnv()
+
+	return err
+}
+
+func (ca *Cagent) CreateDefaultConfigFile(configFilePath string) error {
+	var err error
+	defer func() {
```

Comment:
Is there a reason why this is done using `defer`?

Reply:
Any type of error means there was issue so config may no be valid.

Reply:
I get the error but I wonder _why_ you do this with defer?
Just check normally in the end of the function for an error